src/TranslatableTrait.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace atk4\core;
+
+/**
+ * If a class use this trait, string can be translated calling method translate.
+ */
+trait TranslatableTrait
+{
+    /**
+     * Check this property to see if trait is present in the object.
+     *
+     * @var bool
+     */
+    public $_translatableTrait = true;
+
+    /**
+     * Get the translation of a message in the correct plural form.
+     *
+     * @param string $string  the string to be translated
+     * @param int    $count   the counter used to evaluate the plural
+     *
+     * @return string
+     */
+    public function _(string $string, int $count = 1): string
+    {
+        if (isset($this->app) && method_exists($this->app, '_')) {
+            return $this->app->_($string, $count, 'atk4');
+        }
+
+        return $string;
+    }
+
+    /**
+     * Get the translation of a message in the correct plural form with context domain.
+     *
+     * @param string $context the context domain if exists
+     * @param string $string  the string to be translated
+     * @param int    $count   the counter used to evaluate the plural
+     *
+     * @return string
+     */
+    public function _d(string $context, string $string, int $count = 1): string
+    {
+        if (isset($this->app) && method_exists($this->app, '_')) {
```

Comment:
if we try using `self` here will it give us trait or class ?

Reply:
always class who use the trait, trait as a type not exists

Reply:
that's awesome, can you then extract the namespace and use it as a context instead of `atk` ?